docs/auth.md

```diff
@@ -4,35 +4,225 @@ linkTitle: "Authentication"
 weight: 7
 ---
 -->
-# Authentication
+# Authentication at Run Time
+
+This document describes how Tekton handles authentication when executing
+`TaskRuns` and `PipelineRuns`. Since authentication concepts and processes
+apply to both of those entities in the same manner, this document collectively
+refers to `TaskRuns` and `PipelineRuns` as `Runs` for the sake of brevity.
```

Comment:
This might get confusing with the work @ImJasonH is doing, which introduces a new higher-level concept called "Runs". :(

I think it's fine for now, but maybe worth thinking about.

Reply:
When will Jason's "Runs" be ready? We can update the doc then.

Reply:
Leaving it as-is, we'll revisit once the Jason "Runs" are out. :)